lib/ansible/plugins/connection/persistent.py

```diff
@@ -92,10 +93,12 @@ def _start_connection(self):
         (stdout, stderr) = p.communicate()
         stdin.close()
 
-        if p.returncode == 0:
-            result = json.loads(to_text(stdout, errors='surrogate_then_replace'))
-        else:
-            result = json.loads(to_text(stderr, errors='surrogate_then_replace'))
+        out = stdout if p.returncode == 0 else stderr
+        if br"##RESPONSE##:" not in out:
+            return None
+
+        out = out.split(br"##RESPONSE##:")[1]
+        result = json.loads(to_text(out.rstrip(), errors='surrogate_then_replace'))
 
```

Comment:
Let us print the verbose logs as well? 
`for line in output[0].splitlines():`

Reply:
Fixed